src/ps/scripts.ps1

```diff
@@ -1,76 +1,67 @@
 function GetLatestSlackVersionFolder {
-    $versions = 
-        { [int]($_.Name -replace '.*?-(\d+)\.(\d+)\.(\d+)', '$1') }, 
-        { [int]($_.Name -replace '.*?-(\d+)\.(\d+)\.(\d+)', '$2') }, 
-        { [int]($_.Name -replace '.*?-(\d+)\.(\d+)\.(\d+)', '$3') }
-
-    $latestVersionFolder = Get-ChildItem $env:LOCALAPPDATA\slack\app-* | Sort-Object $versions -Descending | Select-Object -First 1 -ExpandProperty Name
+    $latestVersionFolder = $filename = cmd /c "dir $env:LOCALAPPDATA\slack\app* /ad /on /b" | select -last 1
     return $latestVersionFolder
 }
+$global:latestVersionFolder = GetLatestSlackVersionFolder
```

Comment:
I will try to spend some time reading more about PS scopes. What benefit will this have? GetLatestSlackVersionFolder is only called once at a time in a usage of the script: 1) the dev will use it from StartSlack 2) the user installing it will use it from InstallSlackPatch 3) the user uninstalling from UninstallSlackPatch

Reply:
This was just to show leveraging global vars. No real optimization there, but if you can have couple less lines of code to do the same thing, why not.